sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -279,6 +280,49 @@ case class Ceil(child: Expression) extends UnaryMathExpression(math.ceil, "CEIL"
   override protected def withNewChildInternal(newChild: Expression): Ceil = copy(child = newChild)
 }
 
+object Ceil {
+  def apply(child: Expression, scale: Expression): Expression = {
```

Comment:
It looks an overkill to add a new `Ceil` object. Can we simply put the logic in `CeilExpressionBuilder`?

Reply:
This `Ceil` object serves few unit tests in `MathExpressionsSuite`. I am not sure, we can do the same with only `CeilExpressionBuilder` :
`checkEvaluation(checkDataTypeAndCast(Ceil(doublePi, scale)), doubleResultsCeil(i), EmptyRow)`
Could we ?